### ISSUE TITLE
Create a text-index CLI mode for embedded use cases

### ISSUE THREAD

**Issue by cmdcolin** [GMOD/jbrowse-components#2282]
concept: embedded  indexing of per-track and aggregate text search adapters. justification: embedded doesn't really have a config.json that the jbrowse-web uses

```
jbrowse text-index --file myfile.gff
```

outputs
```
myfile.ix
myfile.ixx
```

if --file specified multiple times it outputs an aggregate

```
aggregate.ix
aggregate.ixx
```

Add docs for embedded. I'd add this to storybook probably, and maybe interlink from config_guide

**Comment by teresam856** [comment]
oclif allows a flag to have multiple values

The command would look like this for multiple files:
``` $ jbrowse text-index --file myfile1.gff myfile2.gff ```
Then the values for the file flag would be `['myfile1.gff', 'myfile2.gff']`.

**Comment by rbuels** [comment]
that sounds fine to me

**Comment by bbimber** [comment]
I was about to open an issue with this question, but see you are ahead of me. In the interim, is there a minimal config.json one could create that would provide enough for the command-line tool to execute? can it have just "tracks", with one track, for example?

**Comment by cmdcolin** [comment]
currently, the minimal config would have tracks and assemblies, like a full config here https://github.com/GMOD/jbrowse-components/blob/main/test_data/volvox/config.json.

the purpose of having it this way is that each assembly gets it's own index.

**Comment by cmdcolin** [comment]
looping back to this issue though, do you think the way we have outlined the solution in the OP here is sufficient for your needs?

**Comment by bbimber** [comment]
> looping back to this issue though, do you think the way we have outlined the solution in the OP here is sufficient for your needs?

If I understand it, the proposal is to support this style command: 
```
jbrowse text-index --file myfile.gff
```

which probably is fine and certainly by far the easiest. I dont know the answer to this, but where that would break down would be if there is configuration present in the track JSON that's relevant. For example, specifying attributes to index or ignore, feature types to ignore, etc. Your defaults as I understand them seem reasonable (i.e. maybe we dont need to specify anything additional), but I'm trying actively dig more into this.

**Comment by cmdcolin** [comment]
@bbimber this is good feedback
 
@teresam856 and I made a PR for the --file and it also allows a config.json with a single assembly so that a the config from @jbrowse/react-linear-genome-view could be supplied

**Comment by cmdcolin** [comment]
Closed by #2303

**Comment by cmdcolin** [comment]
one of the last remaining issues is in #2307 ...hopefully can resolve soon :)
